app/assets/javascripts/whole_pot_calculator.js

```diff
@@ -0,0 +1,212 @@
+(function() {
+  'use strict';
+
+  function WholePotCalculator(income, pot) {
+    this.income = income;
+    this.pot = pot;
+
+    this.standardPersonalAllowance = 10600;
+
+    this.basicRateUpperLimit = 31785;
+    this.basicRateTax = 0.2;
+
+    this.higherRateUpperLimit = 150000;
+    this.higherRateTax = 0.4;
+
+    this.additionalRateTax = 0.45;
+
+    this.taxablePotPortion = 0.75;
+  }
+
+  WholePotCalculator.prototype.round = function(num) {
+    return Math.round(num * 100) / 100;
+  };
+
+  WholePotCalculator.prototype.personalAllowanceFor = function(totalIncome) {
+    if(totalIncome <= 100000) {
```

Comment:
`100000` is a magic number. A constant that revealed what this number means such as `ADJUSTED_NET_INCOME_LIMIT` would be good.